src/input/Binding.zig

```diff
@@ -890,43 +982,50 @@ pub const Action = union(enum) {
         };
     }
 
+    pub const ParseActionErrors = error{InvalidAction} || ParseParameterErrors;
+
+    pub fn parseCLI(input_: ?[]const u8) ParseActionErrors!Action {
+        const input = input_ orelse return error.InvalidFormat;
+        return try parse(input);
+    }
+
     /// Parse an action in the format of "key=value" where key is the
     /// action name and value is the action parameter. The parameter
     /// is optional depending on the action.
-    pub fn parse(input: []const u8) !Action {
+    pub fn parse(input: []const u8) ParseActionErrors!Action {
         // Split our action by colon. A colon may not exist for some
         // actions so it is optional. The part preceding the colon is the
         // action name.
         const colonIdx = std.mem.indexOf(u8, input, ":");
         const action = input[0..(colonIdx orelse input.len)];
+        const param_ = if (colonIdx) |idx| input[idx + 1 ..] else null;
```

Comment:
I think this might change the performance in a subtle way. I think I did the previous approach because we can avoid the branch on `colonIdx` being set if we don't need to care. I'll have to double check.